utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -2,16 +2,81 @@
 #
 # This source file is part of the Swift.org open source project
 #
-# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
 # Licensed under Apache License v2.0 with Runtime Library Exception
 #
 # See https://swift.org/LICENSE.txt for license information
 # See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 #
 # ----------------------------------------------------------------------------
 
-from . import product
+import os
+import platform
 
+from . import product
+from .. import shell
 
 class SwiftPM(product.Product):
-    pass
+    @classmethod
+    def product_source_name(cls):
+        return "swiftpm"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return True
+
+    def run_build_script_helper(self, action, host_target, additional_params=[]):
+        script_path = os.path.join(
+            self.source_dir, 'Utilities', 'bootstrap')
+        toolchain_path = self.install_toolchain_path()
+        swiftc = os.path.join(toolchain_path, "usr", "bin", "swiftc")
+        sbt = os.path.join(toolchain_path, "usr", "bin", "swift-build-tool")
+
+        llbuild_src = os.path.join(
+             os.path.dirname(self.source_dir), "llbuild")
+
+        # FIXME: We require llbuild build directory in order to build. Is
+        # there a better way to get this?
+        build_root = os.path.dirname(self.build_dir)
+        llbuild_build_dir = os.path.join(
+                build_root, '%s-%s' % ("llbuild", host_target))
+
+        helper_cmd = [script_path]
+
+        if action != "build":
+            helper_cmd.append(action)
+
+        if self.is_release():
+            helper_cmd.append("--release")
+
+        helper_cmd += [
+            "--swiftc=" + swiftc,
+            "--sbt=" + sbt,
+            "--build", self.build_dir,
```

Comment:
Any reason why all the other commands are `=` separated and this one is space separated?

Reply:
I don't think so. I'll check.